### PR TITLE
Enhance Share Health Status Verify/ReApply

### PR DESCRIPTION
**Pull request by noah-paige**
### Feature or Bugfix

- Feature

### Detail
- Enhance Share UI for re-apply and verify share health status

### Relates
- #1107 

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by noah-paige** [comment]
Changes:

- [1] Expose health status text + remove tooltip
  - Make health status text visible to user on ShareView to avoid confusion of icons or inconsistency in text size/font

- [2] TODO Ensure auto-refresh after verifying share object

- [3] If item is not shared (`['Revoke_Failed', 'Share_Succeeded']`) then do not expose Item health status but instead show `Item Not Shared`

- [4] For Shares that are in `pendingReApply` - do not allow revokes until re-apply task is finished

- [5] TODO Add Modal to navigate to Share View after verify share on dataset

File: frontend/src/modules/Shares/views/ShareView.js [hunk 1]
```diff
@@ -489,33 +487,11 @@ export function SharedItem(props) {
...
-            </Tooltip>
-          ) : (
-            <Tooltip
-              title={
-                <Typography>{item.healthStatus || 'Undefined'}</Typography>
-              }
-            >
-              <PendingIcon color={'info'} />
-            </Tooltip>
-          )}
```
Comment (dlpzx) [inline frontend/src/modules/Shares/views/ShareView.js:509]: nit: add some space between the tag and tooltip and the text with the verification time. At the moment it is really tight
image

File: frontend/src/modules/Shared/Shares/ShareEditForm.js [hunk 2]
```diff
@@ -12,7 +12,12 @@ import {
   TextField,
   Typography
 } from '@mui/material';
-import { Defaults, Pager, ShareStatus } from '../../../design';
+import {
+  Defaults,
+  Pager,
+  ShareHealthStatus,
+  ShareStatus
+} from '../../../design';
```
Comment (dlpzx) [inline frontend/src/modules/Shared/Shares/ShareEditForm.js:20]: Although it was not introduced in this PR, could you change the relative imports from design to absolute design? 
import 
```
import {
  Defaults,
  Pager,
  ShareHealthStatus,
  ShareStatus
} from 'design';
```

File: frontend/src/modules/Shared/Shares/ShareEditForm.js [hunk 3]
```diff
@@ -173,7 +192,7 @@ const ItemRow = (props) => {
           )}
           {possibleAction === 'Nothing' && (
             <Typography color="textSecondary" variant="subtitle2">
-              Wait until this item is processed
+              Wait until this item is processed and/or re-apply task is complete
```
Comment (dlpzx) [inline frontend/src/modules/Shared/Shares/ShareEditForm.js:195]: Are we enforcing this in the backend?

**Review by dlpzx — APPROVED** [review]
Left some comments, but the feature looks good. I tested it locally.

File: frontend/src/modules/Shared/Shares/ShareEditForm.js (hunk 3 above)
Comment (noah-paige) [reply frontend/src/modules/Shared/Shares/ShareEditForm.js:190]: addressed in latest commit and tested again to ensure revoke still works for healthy share item and throws error for PendingReApply item

For now only enforcing this check on revoke items - the more closer `healthstatus` and `status` get related - the more I am tempted to include transitions and such in the already existing state machine... for now I am leaving as is to not increase complexity / scope of this PR

**Comment by noah-paige** [comment]
@dlpzx The latest commit addressed you comments c9ccb16095e104e72d32366183e40c69f009d41e - take a look and I will merge tomorrow if still all looks good

File: backend/dataall/modules/s3_datasets_shares/db/share_object_repositories.py [hunk 4]
```diff
@@ -683,6 +683,24 @@ def get_share_items_states(session, share_uri, item_uris=None):
             query = query.filter(ShareObjectItem.shareItemUri.in_(item_uris))
         return [item.status for item in query.distinct(ShareObjectItem.status)]
 
+    @staticmethod
+    def get_share_items_health_states(session, share_uri, item_uris=None):
+        query = (
+            session.query(ShareObjectItem)
+            .join(
+                ShareObject,
```
Comment (dlpzx) [inline backend/dataall/modules/s3_datasets_shares/db/share_object_repositories.py]: Do we need the join with the ShareObject? I think we can filter directly on ShareObjectItem.shareUri == share_uri

Comment (noah-paige) [reply backend/dataall/modules/s3_datasets_shares/db/share_object_repositories.py]: removed join - good call out

**Pull request merged**